cluster/gce/gci/configure-helper.sh

```diff
@@ -1540,22 +1547,26 @@ function start-kube-apiserver {
   prepare-log-file "${KUBE_API_SERVER_AUDIT_LOG_PATH:-/var/log/kube-apiserver-audit.log}"
 
   # Calculate variables and assemble the command line.
-  local params="${API_SERVER_TEST_LOG_LEVEL:-"--v=2"} ${APISERVER_TEST_ARGS:-} ${CLOUD_CONFIG_OPT}"
+  #local params="${API_SERVER_TEST_LOG_LEVEL:-"--v=2"} ${APISERVER_TEST_ARGS:-} ${CLOUD_CONFIG_OPT}"
+  local params="--v=9 ${APISERVER_TEST_ARGS:-} ${CLOUD_CONFIG_OPT}"
   params+=" --address=127.0.0.1"
   params+=" --allow-privileged=true"
   params+=" --cloud-provider=gce"
   params+=" --client-ca-file=${CA_CERT_BUNDLE_PATH}"
-  params+=" --etcd-servers=${ETCD_SERVERS:-http://127.0.0.1:2379}"
+
+  if [[ -n "${ETCD_APISERVER_CA_KEY:-}" && -n "${ETCD_APISERVER_CA_CERT:-}" && -n "${ETCD_APISERVER_CLIENT_KEY:-}" && -n "${ETCD_APISERVER_CLIENT_CERT:-}" ]]; then
+      params+=" --etcd-servers=${ETCD_SERVERS:-https://127.0.0.1:2379}"
+      params+=" --etcd-cafile=${ETCD_APISERVER_CA_CERT_PATH}"
+      params+=" --etcd-certfile=${ETCD_APISERVER_CLIENT_CERT_PATH}"
+      params+=" --etcd-keyfile=${ETCD_APISERVER_CLIENT_KEY_PATH}"
+  else
```

Comment:
I think you should log a warning if only some of the required certs & keys are present. I think in most intended cases it should be all-or-none?

Reply:
That's right, I added the warning here.

Reply:
Or crash. No one reads the logs, and will end up with mTLS disabled between API server and etcd which could be dangerous in the future. It's probably better that tests go red.

Reply:
@mikedanese Thanks for your suggestion, I modified the file upon your suggestion and now bazil test failed as below, PTAL:

exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //cluster/gce/cos:go_default_test
-----------------------------------------------------------------------------
--- FAIL: TestEncryptionProviderFlag (0.10s)
    --- FAIL: TestEncryptionProviderFlag/ENCRYPTION_PROVIDER_CONFIG_is_set (0.05s)
        configure_helper_test.go:133: Start kubernetes api-server
            ERROR: One or all of ETCD_APISERVER_CA_KEY, ETCD_APISERVER_CA_CERT, ETCD_APISERVER_SERVER_KEY, ETCD_APISERVER_SERVER_CERT, ETCD_APISERVER_CLIENT_KEY and ETCD_APISERVER_CLIENT_CERT are missing, mTLS between etcd server and kube-apiserver cannot be enabled.
        configure_helper_test.go:134: Failed to run configure-helper.sh: exit status 1
    --- FAIL: TestEncryptionProviderFlag/ENCRYPTION_PROVIDER_CONFIG_is_not_set (0.05s)
        configure_helper_test.go:133: Start kubernetes api-server
            ERROR: One or all of ETCD_APISERVER_CA_KEY, ETCD_APISERVER_CA_CERT, ETCD_APISERVER_SERVER_KEY, ETCD_APISERVER_SERVER_CERT, ETCD_APISERVER_CLIENT_KEY and ETCD_APISERVER_CLIENT_CERT are missing, mTLS between etcd server and kube-apiserver cannot be enabled.
        configure_helper_test.go:134: Failed to run configure-helper.sh: exit status 1
--- FAIL: TestEncryptionProviderConfig (0.04s)
    configure_helper_test.go:133: Start kubernetes api-server
        ERROR: One or all of ETCD_APISERVER_CA_KEY, ETCD_APISERVER_CA_CERT, ETCD_APISERVER_SERVER_KEY, ETCD_APISERVER_SERVER_CERT, ETCD_APISERVER_CLIENT_KEY and ETCD_APISERVER_CLIENT_CERT are missing, mTLS between etcd server and kube-apiserver cannot be enabled.
    configure_helper_test.go:134: Failed to run configure-helper.sh: exit status 1
--- FAIL: TestKMSIntegration (0.08s)
    --- FAIL: TestKMSIntegration/CLOUD_KMS_INTEGRATION_is_set (0.04s)
        configure_helper_test.go:133: Start kubernetes api-server
            ERROR: One or all of ETCD_APISERVER_CA_KEY, ETCD_APISERVER_CA_CERT, ETCD_APISERVER_SERVER_KEY, ETCD_APISERVER_SERVER_CERT, ETCD_APISERVER_CLIENT_KEY and ETCD_APISERVER_CLIENT_CERT are missing, mTLS between etcd server and kube-apiserver cannot be enabled.
        configure_helper_test.go:134: Failed to run configure-helper.sh: exit status 1
    --- FAIL: TestKMSIntegration/CLOUD_KMS_INTEGRATION_is_not_set (0.04s)
        configure_helper_test.go:133: Start kubernetes api-server
            ERROR: One or all of ETCD_APISERVER_CA_KEY, ETCD_APISERVER_CA_CERT, ETCD_APISERVER_SERVER_KEY, ETCD_APISERVER_SERVER_CERT, ETCD_APISERVER_CLIENT_KEY and ETCD_APISERVER_CLIENT_CERT are missing, mTLS between etcd server and kube-apiserver cannot be enabled.
        configure_helper_test.go:134: Failed to run configure-helper.sh: exit status 1
FAIL